router/oauthResponseHandler/oauthResponseHandler.go

```diff
@@ -0,0 +1,178 @@
+		return nil, err
+	}
+	return j, nil
+}
+
+func init() {
+	configBEURL = backendconfig.GetConfigBackendURL()
+	pkgLogger = logger.NewLogger().Child("router").Child("OAuthResponseHandler")
+	loggerNm = "OAuthResponseHandler"
+}
+
+func (authErrHandler *OAuthErrResHandler) Setup() {
+	authErrHandler.logger = pkgLogger
+	authErrHandler.tr = &http.Transport{}
+	//This timeout is kind of modifiable & it seemed like 10 mins for this is too much!
+	authErrHandler.client = &http.Client{Transport: authErrHandler.tr, Timeout: 5 * time.Minute}
+	authErrHandler.oauthErrHandlerReqTimerStat = stats.NewStat("router.processor.oauthErrorHandler_request_time", stats.TimerType)
+	authErrHandler.oauthErrHandlerNetReqTimerStat = stats.NewStat("router.oauthErrorHandler_network_request_time", stats.TimerType)
+}
+
+func (authErrHandler *OAuthErrResHandler) RefreshToken(workspaceId string, accountId string, accessToken string) (statusCode int, respBody string) {
+	authErrHandler.oauthErrHandlerReqTimerStat.Start()
+	if len(strings.TrimSpace(accessToken)) == 0 {
+		return http.StatusBadRequest, `Cannot proceed with refresh token request as accessToken is empty`
+	}
+	authErrHandler.oauthErrHandlerNetReqTimerStat.Start()
+	refreshUrl := fmt.Sprintf("%s/dest/workspaces/%s/accounts/%s/token", configBEURL, workspaceId, accountId)
+	authErrHandler.oauthErrHandlerNetReqTimerStat.End()
+	refTokenBody := RefreshTokenBody{
+		hasExpired:   true,
+		expiredToken: accessToken,
+	}
+	res, err := json.Marshal(refTokenBody)
+	if err != nil {
+		panic(err)
+	}
+	refreshResponse, refreshErr := authErrHandler.client.Post(refreshUrl, "application/json; charset=utf-8", bytes.NewBuffer(res))
+	if refreshErr != nil {
+		return http.StatusBadRequest, err.Error()
+	}
+	statusCode, response := authErrHandler.processResponse(refreshResponse, refreshErr)
+	authErrHandler.oauthErrHandlerReqTimerStat.End()
+	authErrHandler.logger.Debugf("[%s request] :: Refresh token response received : %s", loggerNm, response)
+	return statusCode, response
+}
+
+func (authErrHandler *OAuthErrResHandler) DisableDestination(destination backendconfig.DestinationT, workspaceId string) (statusCode int, respBody string) {
+	authErrHandler.oauthErrHandlerReqTimerStat.Start()
+	destinationId := destination.ID
+	disableURL := fmt.Sprintf("%s/workspaces/%s/destinations/%s/disable", configBEURL, workspaceId, destinationId)
+	req, err := http.NewRequest(http.MethodDelete, disableURL, nil)
+	if err != nil {
+		// Abort on receiving an error in request formation
+		return http.StatusBadRequest, err.Error()
+	}
+	authErrHandler.logger.Debugf("[%s request] :: Disable Request sent : %s", loggerNm)
+	authErrHandler.oauthErrHandlerNetReqTimerStat.Start()
+	res, doErr := authErrHandler.client.Do(req)
+	authErrHandler.oauthErrHandlerNetReqTimerStat.End()
+	if doErr != nil {
+		// Abort on receiving an error
+		return http.StatusBadRequest, err.Error()
+	}
+	statusCode, resp := authErrHandler.processResponse(res, doErr)
```

Comment:
why do we need to pass on error when we are returning it if not nil above

Reply:
Yes We don't need to pass error. Will change it